corehq/apps/registration/utils.py

```diff
@@ -144,9 +146,31 @@ def request_new_domain(request, project_name, is_new_user=True, is_new_sso_user=
     dom_req.domain = new_domain.name
 
     if not settings.ENTERPRISE_MODE:
-        _setup_subscription(new_domain.name, current_user)
+        try:
+            _setup_subscription(new_domain.name, current_user)
+        except Exception as error:
```

Comment:
Can you add a comment explaining why we're catching the generic exception? It's generally considered a smell, so I think it would be good to document why it's appropriate and intentional here.